webapp/templates/jury/analysis/contest_overview.html.twig

```diff
@@ -356,26 +356,47 @@ nv.addGraph(function() {
 // x-axis is contest time
 // y axis is # of submissions
 
+{% set max_submissions_per_minute = 0 %}
+
 var submission_stats = [
   {% for result in ['correct', 'wrong-answer', 'timelimit', 'run-error', 'compiler-error', 'no-output'] %}
   {
     key: "{{result}}",
     color: "{{colors[result]}}",
-    values: [
-      {# TODO: make sure these are actually ordered by submittime #}
-      {# TODO: also make sure these submissions are in the same contest #}
-      [0,0],
-      {% set count = 0 %}
-      {% for submission in submissions | filter(submission => submission.result) %}
-        {% if submission.result == result %}{% set count = count +1 %}{% endif %}
-        [ {{ (submission.submittime - current_contest.starttime)/60.0 }},
-          {{ count }}
-        ],
-        {% endfor %}
-    ]
+    values: []
   },
   {% endfor %}
 ];
+
+var submissions = [
+  {% for submission in submissions %}
+  {
+    result: "{{ submission.result }}",
+    submittime: {{ submission.submittime }},
+    starttime: {{ current_contest.starttime }}
+  }{{ loop.last ? '' : ',' }}
+  {% endfor %}
+];
+
+var contest_duration_minutes = {{ ((current_contest.endtime - current_contest.starttime) / 60)|round }};
+
+submission_stats.forEach(stat => {
+  stat.values = Array.from({ length: contest_duration_minutes }, (_, i) => [i, 0]);
+});
+
+const statMap = submission_stats.reduce((map, stat) => {
+  map[stat.key] = stat;
+  return map;
+}, {});
+
+submissions.forEach(submission => {
+  let submission_minute = Math.round((submission.submittime - submission.starttime) / 60);
```

Comment:
Small detail: this should not be rounded, but the floor taken. A submission at 57 seconcs into the contest is in minute 0, not minute 1. Above `contest_duration_minutes` should be taken the `ceil` though, to make sure that `0 <= submission_minute < contest_duration_minutes` always holds.

Reply:
If a contest lasts 300 minutes, can a submission be made at 299 minutes and 59 seconds? Is it possible to submit during the 300th minute? If not, then should contest_duration_minutes be taken as the floor (rounded down)?

Reply:
Oh, sorry, I understand now. It was my mistake.